File: Solutions/547. Number of Provinces/547.py
class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        n = len(isConnected)
        visited = [0 for _ in range(n)]
        count = 0
        for i in range(n):
            if visited[i] == 0:
                visited[i] = 1
                count += 1
                self.dfs(isConnected, visited, i)
        return count

    def dfs(self,  isConnected: List[List[int]],  visited: List[int], i: int) -> None:
        m = len(isConnected[0])
        for j in range(m):
            if visited[j] == 0 and isConnected[i][j] == 1:
                visited[j] = 1
                self.dfs(isConnected, visited, j)

File: Solutions/547. Number of Provinces/547.java
class Solution {
    public int findCircleNum(int[][] isConnected) {
        int n = isConnected.length;
        int[] visited = new int[n];
        int count = 0;
        for (int i=0;i<n;i++){
            if(visited[i] != 1){
                visited[i] = 1;
                count++;
                dfs(isConnected, visited, i);
            }
        }
        return count;
    }
    public void dfs(int[][] isConnected, int[] visited, int i){
        int m = isConnected[0].length;
        for(int j=0;j<m;j++){
            if(visited[j] != 1 && isConnected[i][j] == 1){
                visited[j] = 1;
                dfs(isConnected, visited, j);
            }
        }
    }
}